Realm/Tests/AsyncTests.mm

```diff
@@ -415,6 +415,8 @@ - (void)testErrorHandling {
     auto token = [IntObject.allObjects addNotificationBlock:^(RLMResults *results, RLMCollectionChange *change, NSError *error) {
         XCTAssertNil(results);
         XCTAssertNotNil(error);
+        // Validate error
+        XCTAssertEqualObjects(error.domain, RLMErrorDomain);
```

Comment:
In general any place where we're checking that something is equal to an exported constant in a test we want to use `XCTAssertEqual()` rather than `XCTAssertEqualObjects()` so that hardcoding the value somewhere rather than using the constant will produce a test failure (at least in debug builds; in release builds string interning will foil that).